gax-java/gax/src/main/java/com/google/api/gax/rpc/EndpointContext.java

```diff
@@ -272,10 +272,6 @@ private String determineUniverseDomain() {
 
     /** Determines the fully resolved endpoint and universe domain values */
     private String determineEndpoint() throws IOException {
-      if (shouldUseS2A()) {
```

Comment:
WDYT of renaming this to `canUseS2A()`, since that's what the result of this function is really indicating. 

For example, it is possible that this function returns true and `canUseDirectPath` in InstantiatingGrpcChannelProvider` returns true, and we therefore end up not using S2A.